src/org/traccar/database/PermissionsManager.java

```diff
@@ -209,6 +215,18 @@ public void checkDeviceReadonly(long userId) throws SecurityException {
         }
     }
 
+    public void checkLimitCommands(long userId) throws SecurityException {
+        if (!isAdmin(userId) && (server.getLimitCommands() || isLimitCommands(userId))) {
+            throw new SecurityException("Account has limit sending commands");
+        }
+    }
+
+    public void checkUserDeviceCommand(long userId, long deviceId, long commandId) throws SecurityException {
+        if (!isAdmin(userId) && Context.getCommandsManager().checkDeviceCommand(deviceId, commandId)) {
```

Comment:
Check if Saved Command linked to device (unwrapped tree)
Admin can send Saved Commands to any device